modules/market/src/main/java/com/opengamma/strata/market/interpolator/ProductPolynomialCurveExtrapolator.java

```diff
@@ -0,0 +1,65 @@
+/**
+ * Copyright (C) 2015 - present by OpenGamma Inc. and the OpenGamma group of companies
+ *
+ * Please see distribution for license.
+ */
+package com.opengamma.strata.market.interpolator;
+
+import java.io.Serializable;
+
+import com.opengamma.strata.math.impl.function.PiecewisePolynomialFunction1D;
+import com.opengamma.strata.math.impl.interpolation.ProductPiecewisePolynomialInterpolator1D;
+
+/**
+ * Extrapolator implementation that uses a polynomial function.
+ * <p>
+ * Given a data set {xValues[i], yValues[i]}, extrapolate {x[i], x[i] * y[i]} by
+ * a polynomial function defined by {@link ProductPiecewisePolynomialInterpolator1D}.
+ * That is, use polynomial coefficients for the extrapolated interval obtained
+ * in {@code ProductPiecewisePolynomialInterpolator1D}.
+ */
+final class ProductPolynomialCurveExtrapolator
+    extends AbstractProductPolynomialCurveExtrapolator
+    implements Serializable {
```

Comment:
Extrapolators are directly referred to in other beans which are serializable. The bound interpoators/extrapolators are not serializable.